controllers/machine.go

```diff
@@ -96,6 +96,10 @@ func (mrc *machineReconcileContext) ensureDependencies() error {
 		return fmt.Errorf("ensuring workflow: %w", err)
 	}
 
+	if err := mrc.setHardwareState(hardware); err != nil {
+		return fmt.Errorf("error setting hardware instance state: %w", err)
+	}
```

Comment:
Since you are calling this inside of tinkerbell machine controller, it won't be able to reconcile the hardware states after the tinkerbell machine is reconciled. So we either have to keep re-queueing until the hardware gets to a expected state or add a [watch](https://book-v1.book.kubebuilder.io/beyond_basics/controller_watches.html) in the machine controller to watch for workflow events.

Reply:
Added!